webextension/background.js

```diff
@@ -20,22 +20,20 @@
 (async function () {
   if(typeof browser.pkcs11 === 'undefined')
     return;
-  var modname = "onepinopenscpkcs11";
-  try {
-    var isInstalled = await browser.pkcs11.isModuleInstalled(modname);
-    if(isInstalled) {
-      console.log("module installed: " + modname);
-      return;
+  async function load(modname) {
+    try {
+      var isInstalled = await browser.pkcs11.isModuleInstalled(modname);
+      if(isInstalled) {
+        console.log("module installed: " + modname);
+        return;
+      }
+      const PKCS11_PUB_READABLE_CERT_FLAG = 0x1<<28;
+      await browser.pkcs11.installModule(modname, PKCS11_PUB_READABLE_CERT_FLAG);
```

Comment:
I add this extension (or I manage to uninstall this extension) if I don't like it. It leaves all the garbage behind.

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/management/onUninstalled

https://github.com/open-eid/firefox-pkcs11-loader/issues/15

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/pkcs11/uninstallModule

Reply:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/management/onUninstalled claims that this is not supported